components/user_profile/index.js

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {makeGetDisplayName} from 'mattermost-redux/selectors/entities/users';
+
+import UserProfile from './user_profile';
+
+function makeMapStateToProps() {
+    const getDisplayName = makeGetDisplayName();
+
+    return (state, ownProps) => {
+        return {
+            displayName: getDisplayName(state, ownProps.userId),
```

Comment:
Accessing `ownProps` within a `mapStateToProps` forces a somewhat slower execution path within react-redux, but it's often easy to avoid by accessing the property from the factory function and assuming that it's not going to change "in place", but would trigger a newly mounted component instead.

Does it makes sense to leverage that pattern here? (i.e. is it safe to assume that the `UserProfile` widget won't have the `userId` change in place?)

The pattern would look like:
```js
function makeMapStateToProps(initialState, initialOwnProps) {
    const getDisplayName = makeGetDisplayName();
    const userId = initialOwnProps.userId;

     return (state) => {
        return {
            displayName: getDisplayName(state, userId),
        };
    };
}
```

Reply:
I read that as well from your report and I thought we'll tackle it separately.  Sure, we can leverage it here since it won't expect change in the userId :)

After implementing it, I can't find any difference in performance metrics (especially on JS scripting) for the before and after.  Maybe we'll have performance improvement once other components are updated with such pattern?